Homework 2.2

vargad opened the pull request.

vargad committed:
Homework 2.2

maciekgajewski commented on course5/homework2.2/homework2.2.cc:
```diff
@@ -0,0 +1,46 @@
+float newton_sqrt(float number, float guess, FuncInspectMetadata &metadata)
+{
+    metadata.ptr_to_local.push_back(&guess);
+    if (std::fabs(guess*guess - number) < EPSILON) return guess;
```
Use std::abs instead of std::fabs

maciekgajewski commented on course5/homework2.2/homework2.2.cc:
```diff
@@ -0,0 +1,46 @@
+#include <cstring>
+#include <vector>
+
+static const float EPSILON = 1.0e-8;
```
this constant is local to a function, should be defined inside of the function, to not pollute the namespace.

maciekgajewski reviewed (commented):
Good

Commit:
Move EPSILON to the function, so it does not pollute the global namespace

Commit:
Use std::abs

Interesting that std::abs and std::fabs has the same overloads, so it
doesn't really matter, I guess std::fabs is kept for compatibility.

Commit:
No need to explicitly add -std=c++14
